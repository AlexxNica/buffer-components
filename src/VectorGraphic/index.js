import React, { PropTypes } from 'react';

const VectorGraphic = ({ children, height, width, viewWidth, viewHeight }) => {
  if (!viewWidth) {
    viewWidth = width;
  }
  if (!viewHeight) {
    viewHeight = height;
  }
  return (
    <svg
      height={height}
      version={'1.1'}
      viewBox={`0 0 ${viewWidth} ${viewHeight}`}
      width={width}
      xmlns={'http://www.w3.org/2000/svg'}
    >
      {children}
    </svg>
  );
};


VectorGraphic.propTypes = {
  children: PropTypes.node.isRequired,
  height: PropTypes.number,
  width: PropTypes.number,
  viewHeight: PropTypes.number,
  viewWidth: PropTypes.number,
};

VectorGraphic.defaultProps = {
  height: 20,
  width: 20,
};

export default VectorGraphic;
