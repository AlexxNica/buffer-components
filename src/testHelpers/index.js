export const renderAndCheckSnapshot = (component) => {
  const renderedComponent = renderer.create(component);
  const tree = renderedComponent.toJSON();
  expect(tree).toMatchSnapshot();
};
