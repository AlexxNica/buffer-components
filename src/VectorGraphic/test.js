jest.unmock('./index');
import React from 'react';
import { renderAndCheckSnapshot } from '../testHelpers';
import VectorGraphic from './index';

describe('VectorGraphic', () => {
  it('VectorGraphic component', () => {
    renderAndCheckSnapshot(
      <VectorGraphic>
        <circle cx="50" cy="50" r="40" stroke="black" strokeWidth="3" fill="red" />
      </VectorGraphic>
    );
  });

  it('VectorGraphic w/ height', () => {
    renderAndCheckSnapshot(
      <VectorGraphic
        height={10}
      >
        <circle cx="50" cy="50" r="40" stroke="black" strokeWidth="3" fill="red" />
      </VectorGraphic>
    );
  });

  it('VectorGraphic w/ width', () => {
    renderAndCheckSnapshot(
      <VectorGraphic
        width={10}
      >
        <circle cx="50" cy="50" r="40" stroke="black" strokeWidth="3" fill="red" />
      </VectorGraphic>
    );
  });

  it('VectorGraphic w/ viewHeight', () => {
    renderAndCheckSnapshot(
      <VectorGraphic
        viewHeight={10}
      >
        <circle cx="50" cy="50" r="40" stroke="black" strokeWidth="3" fill="red" />
      </VectorGraphic>
    );
  });

  it('VectorGraphic w/ viewWidth', () => {
    renderAndCheckSnapshot(
      <VectorGraphic
        viewWidth={10}
      >
        <circle cx="50" cy="50" r="40" stroke="black" strokeWidth="3" fill="red" />
      </VectorGraphic>
    );
  });
});
