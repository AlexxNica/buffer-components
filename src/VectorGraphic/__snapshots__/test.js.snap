exports[`VectorGraphic VectorGraphic component 1`] = `
<svg
  height={20}
  version="1.1"
  viewBox="0 0 20 20"
  width={20}
  xmlns="http://www.w3.org/2000/svg">
  <circle
    cx="50"
    cy="50"
    fill="red"
    r="40"
    stroke="black"
    strokeWidth="3" />
</svg>
`;

exports[`VectorGraphic VectorGraphic w/ height 1`] = `
<svg
  height={10}
  version="1.1"
  viewBox="0 0 20 10"
  width={20}
  xmlns="http://www.w3.org/2000/svg">
  <circle
    cx="50"
    cy="50"
    fill="red"
    r="40"
    stroke="black"
    strokeWidth="3" />
</svg>
`;

exports[`VectorGraphic VectorGraphic w/ viewHeight 1`] = `
<svg
  height={20}
  version="1.1"
  viewBox="0 0 20 10"
  width={20}
  xmlns="http://www.w3.org/2000/svg">
  <circle
    cx="50"
    cy="50"
    fill="red"
    r="40"
    stroke="black"
    strokeWidth="3" />
</svg>
`;

exports[`VectorGraphic VectorGraphic w/ viewWidth 1`] = `
<svg
  height={20}
  version="1.1"
  viewBox="0 0 10 20"
  width={20}
  xmlns="http://www.w3.org/2000/svg">
  <circle
    cx="50"
    cy="50"
    fill="red"
    r="40"
    stroke="black"
    strokeWidth="3" />
</svg>
`;

exports[`VectorGraphic VectorGraphic w/ width 1`] = `
<svg
  height={20}
  version="1.1"
  viewBox="0 0 10 20"
  width={10}
  xmlns="http://www.w3.org/2000/svg">
  <circle
    cx="50"
    cy="50"
    fill="red"
    r="40"
    stroke="black"
    strokeWidth="3" />
</svg>
`;
